### PR TITLE
fix: pruneancient freeze from the previous position when the first time

### PR DESCRIPTION
**Pull request by jingjunLi**
### Description

The first time online pruning is enabled, the freeze() process starts from 0, leading to a large number of log.Error("Canonical hash missing, can't freeze", "number", f.frozen) logs.

image

In the freezer's database, record FrozenOfAncientFreezer. When online pruning is enabled for the first time, it will continue to freeze from the previous position.

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

**Review by joeylichang — APPROVED** [review]

**Review by zzzckck — APPROVED** [review]

File: core/rawdb/prunedfreezer.go [hunk 1]
```diff
@@ -68,20 +70,47 @@ func newPrunedFreezer(datadir string, db ethdb.KeyValueStore, offset uint64) (*p
...
+		hidden := table.itemHidden.Load()
+		if hidden > tail {
+			tail = hidden
 		}
 		table.Close()
 	}
-	log.Info("Read ancientdb item counts", "items", min)
-	offset += min
+	items := head - tail
+	log.Info("Read ancientdb item counts", "items", items)
```
Comment (joeylichang) [inline core/rawdb/prunedfreezer.go:112]: add 'head' and 'tail' info ?

**Review by joeylichang — APPROVED** [review]

**Comment by zzzckck** [comment]
May double check this code, see if it is reasonable:
https://github.com/bnb-chain/bsc/blob/v1.4.11/core/rawdb/prunedfreezer.go#L84-L90

File: core/rawdb/prunedfreezer.go [hunk 2]
```diff
@@ -68,20 +70,40 @@ func newPrunedFreezer(datadir string, db ethdb.KeyValueStore, offset uint64) (*p
...
-	log.Info("Read ancientdb item counts", "items", min)
-	offset += min
+
+	// If minItems is non-zero, it indicates that the chain freezer was previously enabled, and we should use minItems as the current frozen value.
+	// If minItems is zero, it indicates that the pruneAncient was previously enabled, and we should continue using frozen
+	//	(retrieved from CurrentAncientFreezer) as the current frozen value.
+	if minItems != 0 {
+		offset = minItems
+	}
+	log.Info("Read ancientdb item counts", "items", minItems, "offset", offset)
```
Comment (zzzckck) [inline core/rawdb/prunedfreezer.go:106]:
```
frozenInDB := ReadFrozenOfAncientFreezer(f.db)
max := 0
if frozenInDB > offset {
    max = frozenInDB
} else {
    max = offset
}
atomic.StoreUint64(&f.frozen, max)
```

File: core/rawdb/prunedfreezer.go [hunk 3]
```diff
@@ -68,20 +70,40 @@ func newPrunedFreezer(datadir string, db ethdb.KeyValueStore, offset uint64) (*p
...
+			minItems = items
 		}
 		table.Close()
 	}
-	log.Info("Read ancientdb item counts", "items", min)
-	offset += min
+
+	// If minItems is non-zero, it indicates that the chain freezer was previously enabled, and we should use minItems as the current frozen value.
+	// If minItems is zero, it indicates that the pruneAncient was previously enabled, and we should continue using frozen
+	//	(retrieved from CurrentAncientFreezer) as the current frozen value.
```
Comment (zzzckck) [inline core/rawdb/prunedfreezer.go:102]: remove line#71, and update:
```
       offset := minItems
	if offset == 0 {
               // no item in ancientDB, init `offset` to the `f.frozen`
                offset = atomic.LoadUint64(&f.frozen)
        }


```

**Review by zzzckck — APPROVED** [review]

**Review by joeylichang — APPROVED** [review]

**Pull request merged**